internal/db/reset/templates/drop.sql

```diff
@@ -60,6 +60,13 @@ begin
     select *
     from pg_type t
     where t.typnamespace::regnamespace::name = 'public'
+    order by (
+      select count(d.objid)
+      from pg_depend d
+        left join pg_class c on d.refclassid = c.oid
+      where t.oid = d.objid
+        and c.relname = 'pg_type'
+    )
```

Comment:
```suggestion
      and typtype != 'b'
```

Filter out any base types as discussed.